### ISSUE TITLE
Libc: improve passwd database support

### ISSUE THREAD

**Issue by ehmry** [genodelabs/genode#2919]
The libc needs to make a better effort at providing some information about "users". The present dummy for ``getpwuid_r`` is broken because it fails to fill out a struct with pointers that application code will blindly dereference. Having better dummies here is important because for our libc `getenv("HOME")` cannot be expected to return anything.

This issue is related to using libssh for an ssh client.

**Comment by ehmry** [comment]
Rebased against master on my `issue2919_getpwent` branch. I rewrote the Libc config accessor to simple return an `Xml_node`. I had a strange accessor before because I think I tried to make it thread safe, but the libc config ROM is never updated, so its not a concern of mine anymore.

**Comment by chelmuth** [comment]
@ehmry fcc154f46efc433430460e8eb47e0d45f94c1e1f breaks `make run/stdcxx KERNEL=linux` (please consider to test together with fixup 69b771daac5bffc6012ad4cd21cb6951f3ce8e22)

**Comment by chelmuth** [comment]
`make run/vfs_import KERNEL=linux` also chokes on the commit mentioned above.

**Comment by ehmry** [comment]
I have a fix for stdcxx at 15be3b53fb344bde9faaa9f71a1801a34503db5a, the problem with noux I suspect may be `fsync(0);`.

**Comment by nfeske** [comment]
The noux patch does not look right to me. The lifetime of the I/O channels is managed via reference-counting shared pointers. So the correct way would be dropping the pointers by assigning invalid (default-constructed) 'Shared_pointer` values to the `_channel_0` ... `_channel_2` variables. To check that this triggers the desired destruction, it might good to have a look at the `count()` value of the pointer, to see if the code removes the last reference.

**Comment by ehmry** [comment]
@nfeske yes, assignment is suffcient to close these channels, a better patch at c93cba4e28507c959d1a5e05211e65faa476dd55.
